docs/composability.md

```diff
@@ -0,0 +1,24 @@
+# Building a Clean, Readable Distributed LLM
+One of the main goals for TorchTitan was to provide a version of distributed LLM that was not only high performance, but utilized native pytorch techniques and readable code.  The challenge is how to compose together so many individual library components (FSDP, TP, PP, FP8, Compile, DCP, ...) just to name a few, and avoid having to make too many changes to the model guts in the process.  A lot of the work is behind the scenes, designing individual components to make fewer assumptions, use common abstractions (e.g. DTensor) and generally 'get along'.  But we found a few tweaks to the model code invaluable as well, and wanted to share those changes and the rationale for them.
+
+
+
+# Making the model "pipeline friendly"
+Whether you plan to manually edit your model code, or use techniques like tracing to extract model chunks, you will have to construct nn.Module objects representing the portion of the model that runs on a given pipeline stage.  Regardless of how you approach this, a few changes to the original model code can go a long way to making this process easier.
```

Comment:
nit: we can set some overall context in the beginning of the section:

When applying Pipeline Parallelism to the model. whether ...